cmd/src/server.rs

```diff
@@ -149,7 +149,7 @@ struct TiKVServer<ER: RaftEngine> {
 
 struct TiKVEngines<ER: RaftEngine> {
     engines: Engines<RocksEngine, ER>,
-    store_meta: Arc<Mutex<StoreMeta>>,
+    store_meta: Arc<RwLock<StoreMeta>>,
```

Comment:
`Mutex` is used on purpose. Generally, `Mutex` is faster than `RwLock` unless there are many read requests and write requests can hold the lock too long.

Reply:
But there could be a lot of lock contention when a region split after we introduce these changes.

Reply:
Mutex works very good under high contention.

Reply:
🤔 Will benchmark for it.

Reply:
Benchmarks show that mutex works obviously better than rwlock. Will revert the changes.